lib/experimental/fonts-api/class-wp-fonts-resolver.php

```diff
@@ -219,7 +214,12 @@ private static function get_settings() {
 			);
 
 			// Make sure there are no duplicates.
-			$settings['typography']['fontFamilies'] = array_unique( $settings['typography']['fontFamilies'] );
+			$settings['typography']['fontFamilies'] = array_unique( $settings['typography']['fontFamilies'], SORT_REGULAR );
+
+			// The font families from settings might become null after running the `array_unique`.
+			if ( ! isset( $settings['typography']['fontFamilies']['theme'] ) || ! is_array( $settings['typography']['fontFamilies']['theme'] ) ) {
```

Comment:
This check should be before the `array_merge()` not after it? Also ideally outside of the loop, it's pointless to repeat it all the time.

Reply:
The `$settings['typography']['fontFamilies']['theme']` is removed after the `array_unique` call, so I think we can also do it here to ensure the `$settings['typography']['fontFamilies']['theme']` is set.

I agreed the check should be before the `array_merge()` but we still need to set the value outside the loop again after the last iteration to ensure the value is set. I think the better approach is to move the `array_unique` call outside the loop if possible.

Reply:
@arthur791004 
Could you provide a code snippet that would show how `array_unique()` removes `$settings['typography']['fontFamilies']['theme']` ?
I've asked this in the previous PR, and I got no reply.

You could modify [this](https://3v4l.org/SSATQ#v8.0.0) snippet and post a link to the new snippet here.
IMO, `array_unique()` only removes duplicate values. `array_unique()` doesn't set values to `null`.

Reply:
Here is it: https://3v4l.org/ZiPOT#v8.0.0, and I also mentioned the example here, https://github.com/WordPress/gutenberg/pull/56067#issuecomment-1808500788

Reply:
@arthur791004, thank you for expanding on this so I can better understand the proposed idea.
I see your point with this PR.

> When you access the value after that, the value doesn't exist anymore and returns NULL.

Yes. And if an array key doesn't exist, it shouldn't be accessed directly, as accessing an undefined array element generally results [in a PHP warning](https://3v4l.org/5XDZd).

> Maybe this is better than null 😅.

I respectfully disagree with that.
In the general PHP context, it's correct to use both `NULL` and `null` as it makes no difference (code-wise).

In the WordPress context, `null` is being used in all the `*.php` files and in [the PHP coding standards](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/). 😅